samples/cellular/modem_shell/src/sock/sock.c

```diff
@@ -933,7 +948,22 @@ int sock_send_data(
 			if (!sock_send_buffer_calloc(socket_info, data_out_length)) {
 				return -ENOMEM;
 			}
-			memcpy(socket_info->send_buffer, data_out, data_out_length);
+
+			if (packet_number_prefix) {
+				/* Make room for packet buffer prefix */
+				if (data_format_hex) {
+					mosh_error(
+						"Hexadecimal format cannot be used with "
+						"packet number prefix");
+					return -EINVAL;
+				}
+
+				sprintf(socket_info->send_buffer, "0000%s", data_out);
```

Comment:
The length of the send buffer is `data_out_length`, so this writes past the end of the buffer, right?

Reply:
Yep, this is something I was supposed to double check and seems I remembered the buffer allocation wrong.